File: benchmarks/lua/table/array_access.lua
-- Array access benchmark
-- Create array, sum elements, reverse in-place, sum again

local N = tonumber(arg and arg[1]) or 20000000
collectgarbage("collect")

local arr = {}
for i = 1, N do
    arr[i] = i % 100
end

-- First sum
local sum1 = 0
for i = 1, N do
    sum1 = sum1 + arr[i]
end

-- Reverse in-place
local half = math.floor(N / 2)
for i = 1, half do
    local j = N - i + 1
    arr[i], arr[j] = arr[j], arr[i]
end

-- Second sum
local sum2 = 0
for i = 1, N do
    sum2 = sum2 + arr[i]
end

print(sum1)
print(sum2)

File: benchmarks/js/object/array_access.js
// Array access benchmark
// Create array, sum elements, reverse in-place, sum again

var print = typeof console !== 'undefined' ? console.log.bind(console) : globalThis.print;
var DEFAULT = 20000000;
var N = parseInt(typeof scriptArgs !== 'undefined' ? scriptArgs[1] : (typeof process !== 'undefined' ? process.argv[2] : '0')) || DEFAULT;

var arr = new Array(N);
for (var i = 0; i < N; i++) {
    arr[i] = (i + 1) % 100;
}

// First sum
var sum1 = 0;
for (var i = 0; i < N; i++) {
    sum1 += arr[i];
}

// Reverse in-place
var half = Math.floor(N / 2);
for (var i = 0; i < half; i++) {
    var j = N - 1 - i;
    var tmp = arr[i];
    arr[i] = arr[j];
    arr[j] = tmp;
}

// Second sum
var sum2 = 0;
for (var i = 0; i < N; i++) {
    sum2 += arr[i];
}

print(sum1);
print(sum2);
